bin/compile

```diff
@@ -217,18 +214,21 @@ else
   CACHED_PYTHON_STACK=$STACK
 fi
 
-export CACHED_PYTHON_STACK
-
 # Pipenv Python version support.
 # Detect the version of Python requested from a Pipfile (e.g. python_version or python_full_version).
 # Convert it to a runtime.txt file.
 
 # shellcheck source=bin/steps/pipenv-python-version
 source "$BIN_DIR/steps/pipenv-python-version"
 
-# If no runtime was provided by the user, assume the default Python runtime version.
-if [ ! -f runtime.txt ]; then
-  echo "$DEFAULT_PYTHON_VERSION" > runtime.txt
+if [[ -f runtime.txt ]]; then
```

Comment:
I wonder what will be the result of this new logic for the end-user...

Reply:
I think it should be OK, better even..

My understanding is that when no runtime is specified, this will prevent accidental python upgrade on existing(deployed) apps when `DEFAULT_PYTHON_VERSION` changes (as a result of on-prem/buildpack upgrades)

Reply:
Sounds good.